### PR TITLE
Initial labstack echo instrumentation with otel-go

### PR DESCRIPTION
**Pull request by toffentoffen**
This is the initial opentelemetry implementation for the [echo](https://github.com/labstack/echo) web framework.

### CODE REVIEW

File: plugins/labstack/echo/echo.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		cfg := Config{}
+		for _, opt := range opts {
+			opt(&cfg)
+		}
+		if cfg.Tracer == nil {
+			cfg.Tracer = otelglobal.Tracer(tracerName)
+		}
+		if cfg.Propagators == nil {
+			cfg.Propagators = otelglobal.Propagators()
+		}
```
[the remark this answers is not in the document]
Comment (toffentoffen) [reply plugins/labstack/echo/echo.go:46]: The returned `echo.MiddlewareFunc` is invoke with every request.
I put the initialisation block there following a bit https://github.com/DataDog/dd-trace-go/blob/v1/contrib/labstack/echo/echotrace.go#L23, but your suggestion are totally right.

**Comment by toffentoffen** [comment]
@Aneurysm9 I think that the `Blocked on CLA` label can be removed, I think it is sorted out now.
Also who else needs to approve the review to get it merged? Thanks in advance for your time.

**Comment by Aneurysm9** [comment]
> @Aneurysm9 I think that the `Blocked on CLA` label can be removed, I think it is sorted out now.

Yup, the label has been removed.  

> Also who else needs to approve the review to get it merged? Thanks in advance for your time.

Someone else from @open-telemetry/go-approvers needs to approve and then @open-telemetry/go-maintainers can merge.

**Comment by jmacd** [comment]
Thank you @toffentoffen!

**Pull request merged**